### DISCUSSION TITLE
WinAppSDK 1.5, Packaged with WAP project and CommunityToolkit 8.1

### DISCUSSION THREAD

**Discussion by optsing** [CommunityToolkit/Windows#485] [Q&A]
After updating CommunityToolkit (Animations, Converters, Extensions) from 8.0.240109 to 8.1.240821, the packaged version of the application does not launch, it gives the error “Cannot create instance of type ‘Microsoft.UI.Xaml.Controls.XamlControlsResources’” in App.xaml. But the unpackaged application works fine. 
Target system version is 10.0.22621.0 in the original project, in wapproj and in appxmanifest, WinAppSDK version is 1.5.240802000.

Is it now required to set any specific parameters for wapproj or is this an issue?

**Comment by Arlodotexe** [comment]
Please make sure your `TargetFramework` is updated in order to upgrade to 8.1.

- If you were targeting `net6.0-windows10.0.18362`, you would target `net6.0-windows10.0.22621` instead now. Similar for `net7.0-windows*` and `net8.0-windows*`.
- `SupportedOSVersion` or `TargetPlatformMinVersion` can stay the same, as low as `17763`.

This is detailed at the top of the [release notes](https://github.com/CommunityToolkit/Windows/releases/tag/v8.1.240821) and [blog announcement](https://devblogs.microsoft.com/ifdef-windows/announcing-windows-community-toolkit-v8-1/).

**Reply by optsing** [reply]
These requirements are fulfilled. The unpackaged application works fine. Packaged via wapproj gives an error on startup.

```Exception thrown at 0x00007FFEE6BE6D9A (KernelBase.dll) in App.exe: WinRT originate error - 0x8001010E : 'The application called an interface that was marshalled for a different thread.'.
Exception thrown at 0x00007FFEE6BE6D9A in App.exe: Microsoft C++ exception: winrt::hresult_wrong_thread at memory location 0x00000028CF57D270.
Exception thrown at 0x00007FFEE6BE6D9A in App.exe: Microsoft C++ exception: [rethrow] at memory location 0x0000000000000000.
Exception thrown: 'System.Runtime.InteropServices.COMException' in System.Private.CoreLib.dll
Exception thrown at 0x00007FFEE6BE6D9A (KernelBase.dll) in App.exe: WinRT originate error - 0x8001010E : 'The application called an interface that was marshalled for a different thread.'.
Microsoft.ui.xaml.dll!00007FFDCDBFA876: 8001010E - RPC_E_WRONG_THREAD
Exception thrown at 0x00007FFEE6BE6D9A (KernelBase.dll) in App.exe: WinRT originate error - 0x802B000A : 'Cannot create instance of type 'Microsoft.UI.Xaml.Controls.XamlControlsResources' [Line: 8 Position: 40]'.```

**Reply by Arlodotexe** [reply]
Thanks for reporting, we're looking into this.
